cpp/src/gandiva/gdv_string_function_stubs.cc

```diff
@@ -264,6 +265,304 @@ const char* gdv_fn_lower_utf8(int64_t context, const char* data, int32_t data_le
   return out;
 }
 
+// converts utf8 to utf16 encoding
+GANDIVA_EXPORT
+const char* gdv_fn_encode(int64_t context, const char* data, int32_t data_len,
+                          const char* charset, int32_t charset_len, int32_t* out_len) {
+  if (data_len == 0) {
+    *out_len = 0;
+    return "";
+  }
+
+  char* out =
+      reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, 32 * data_len));
+  if (out == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
+    *out_len = 0;
+    return "";
+  }
+
+  int32_t char_len;
+  uint32_t char_codepoint;
+  int32_t y = 5;
+  int32_t z = 9;
+
+  std::string charset_str(charset, charset_len);
+
+  if (boost::iequals(charset_str, "UTF-16BE")) {
+    char* tmp = out;
+
+    int ind_w = 0;
+
+    for (int32_t i = 0; i < data_len; i += char_len) {
+      char_len = gdv_fn_utf8_char_length(data[i]);
+
+      if (char_len == 1) {
+        int32_t input = static_cast<int32_t>(data[i]);
+
+        auto num_bytes = snprintf(tmp + ind_w, y, "%.4X", input);
```

Comment:
I though this would output an actual utf16 (binary) string, not hex-encoded utf16 codeunits? Also, calling `snprintf` each time will be quite slow...

Reply:
What would you suggest to do?

Reply:
Because it would be easier to convert it to other format as it in hex format so no preprocessing further would be required.

Reply:
Which kind of other format? In which situation do you want hex-encoded utf16 rather than raw utf16?
If someone wants raw utf16, do they have to hex-decode the result that was hex-encoded here? It sounds extremely wasteful...

Reply:
ISO formats, utf8 and ascii also sometimes.

Reply:
This doesn't answer the questions above:
> In which situation do you want hex-encoded utf16 rather than raw utf16? If someone wants raw utf16, do they have to hex-decode the result that was hex-encoded here? It sounds extremely wasteful...